### ISSUE TITLE
Add Opaque as default type in kubectl create secret .

### ISSUE THREAD

**Issue by Ritikaa96** [kubernetes/kubernetes#127914]
### What happened?

Reference : https://github.com/kubernetes/kubernetes/pull/120337/files#r1408525014
Opaque is not added as default type in code as we see empty string when stating type in help message.
```shell
Options:
...
    --type='':
	The type of secret to create
...
```

### What did you expect to happen?

Options:
```shell
...
    --type='Opaque':
	The type of secret to create
...
```

### How can we reproduce it (as minimally and precisely as possible)?

```shell
kubectl create secret generic --help
```

### Anything else we need to know?

NONE

### Kubernetes version

<details>

```console
$ kubectl version
```

</details>

### Cloud provider

<details>
any
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here
</details>


### Install tools

<details>

</details>


### Container runtime (CRI) and version (if applicable)

<details>

</details>


### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by ardaguclu** [comment]
This would be a breaking change because now we are changing the default to opaque?. What is the current default?.

**Comment by Lokak07** [comment]
@ardaguclu I dont think they are talking about making the change to the feature. but the help documentation showing up as --type='':

when we use **kubectl create secret generic --help**

@Ritikaa96 I can reproduce the issue ?? as per discussion on https://github.com/kubernetes/kubernetes/pull/120337/files#r1408525014

**Comment by Ritikaa96** [comment]
@Lokak07 , you are right. Opaque is default but we needed to show it in the help message as well which currently is not seen. 
This change is for that only @ardaguclu , so no breaking change

**Comment by SayakMukhopadhyay** [comment]
I looked into it and it seems like this is an issue with other options and commands too. For eg. `kubectl create service clusterip --help` gives
```
Options:
    --allow-missing-template-keys=true:
        If true, ignore any errors in templates when a field or map key is missing in the template. Only applies to
        golang and jsonpath output formats.

    --clusterip='':
        Assign your own ClusterIP or set to 'None' for a 'headless' service (no loadbalancing).

    --dry-run='none':
        Must be "none", "server", or "client". If client strategy, only print the object that would be sent, without
        sending it. If server strategy, submit server-side request without persisting the resource.
```
Notice how `--clusterip` also doesn't have a default value set whereas `--dry-run` and `--allow-missing-template-keys` does have default values.

**Comment by SayakMukhopadhyay** [comment]
And the root cause of this issue is due to the use of `o.type` instead of a hardcoded default. That means if a value isn't passed, it's just shows empty in the help.

https://github.com/kubernetes/kubernetes/blob/856bb5c8f266f5276f1a576f47be622d7cb384e7/staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret.go#L164

I am not sure if this is intended cause the same variable is used for both referencing the value and passing the default in all other commands.

**Comment by Ritikaa96** [comment]
It is definitely not intended. but we can make a change in the newOption function and get it correct. I'll be happy to do it once the triage is accepted for this one.

**Comment by Ritikaa96** [comment]
Waiting for more discussion on this from community.

**Comment by Ritikaa96** [comment]
This PR is waiting for further reviews.
